RocketChat/Rocket.Chat.ReactNative
# [FIX] Support chats order for older versions of the server

## Proposed changes

On #2763, we started using the new way of sorting front-end is using, but we dropped support for older versions of the server. All we need to do is check for the current `serverVersion`.

## Issue(s)	

Closes #2692
Closes #2781 

## How to test or reproduce


In order to test this you should be logged in from Web client and the app. In all this cases (with a small exception in discussions creation [*] ) chat order should be the same in Web client and the app

1. Log in
1.1 Login from the Web client
1.2 Login from the app

2. Rooms
2.1 Create a room from the Web client
2.2 Send a message from the Web client
2.3 Leave the room
2.4 Create a new room
2.5 Go back to the old room
2.6 Create a thread and send a message

3. Discussions
3.1 Create a room from the Web client
3.2 Create a discussion in that room from the Web client

## Screenshots

## Types of changes




## Checklist



## Further comments

- This PR requires #2922 to be merged in order to apply the simplified server comparison
- [*] Instead of moving the room as the first chat it always moves the discussion as the first; not a real issue because it does moves the discussion and the room to be the first chats, but just to take in note.
